Can't compile nrfconnect lighting app

#### Problem
When I follow the steps in `examples/lighting-app/nrfconnect/README.md` to get nrfconnect 1.4.0, update deps (like zephyr), and then `west build -b nrf52840dk_nrf52840` I get build errors like so inside zephyr due to warnings that become fatal with #4528:
```
tools/nrfconnect/zephyr/subsys/shell/shell_fprintf.c: In function 'shell_fprintf_buffer_flush':
tools/nrfconnect/zephyr/subsys/shell/shell_fprintf.c:47:53: error: pointer targets in passing argument 2 of 'sh_fprintf->fwrite' differ in signedness [-Werror=pointer-sign]
   47 |  sh_fprintf->fwrite(sh_fprintf->user_ctx, sh_fprintf->buffer,
      |                                           ~~~~~~~~~~^~~~~~~~
      |                                                     |
      |                                                     uint8_t * const {aka unsigned char * const}
tools/nrfconnect/zephyr/subsys/shell/shell_fprintf.c:47:53: note: expected 'const char *' but argument is of type 'uint8_t * const' {aka 'unsigned char * const'}
....
tools/nrfconnect/zephyr/subsys/shell/shell_help.c: In function 'help_item_print':
tools/nrfconnect/zephyr/subsys/shell/shell_help.c:113:37: error: pointer targets in passing argument 1 of 'strlen' differ in signedness [-Werror=pointer-sign]
  113 |  const uint16_t offset = 2 * strlen(tabulator) + item_name_width + 1;
      |                                     ^~~~~~~~~
      |                                     |
      |                                     const uint8_t * {aka const unsigned char *}
In file included from tools/nrfconnect/zephyr/include/sys/ring_buffer.h:16,
                 from tools/nrfconnect/zephyr/include/shell/shell_history.h:13,
                 from tools/nrfconnect/zephyr/include/shell/shell.h:12,
                 from tools/nrfconnect/zephyr/subsys/shell/shell_ops.h:10,
                 from tools/nrfconnect/zephyr/subsys/shell/shell_help.c:7:
tools/gcc-arm-none-eabi-9-2020-q2-update/arm-none-eabi/include/string.h:41:17: note: expected 'const char *' but argument is of type 'const uint8_t *' {aka 'const unsigned char *'}
   41 | size_t  strlen (const char *);
      |                 ^~~~~~~~~~~~
cc1: all warnings being treated as errors
```
etc, etc.  I assume this compiles in our CI somehow, but I'm not sure what the difference is...

@Damian-Nordic 

 #### Proposed Solution
Either turn off -Werror for these files or figure out how to get them to be warning-free.

--- a/examples/lighting-app/nrfconnect/CMakeLists.txt
+++ b/examples/lighting-app/nrfconnect/CMakeLists.txt
@@ -27,8 +27,6 @@ find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
 
 project(chip-nrf52840-lighting-example)
 
-zephyr_compile_options(-Werror)
-
 target_include_directories(app PRIVATE
                            main/include
                            ${LIGHTING_COMMON}

--- a/examples/lock-app/nrfconnect/CMakeLists.txt
+++ b/examples/lock-app/nrfconnect/CMakeLists.txt
@@ -27,8 +27,6 @@ find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
 
 project(chip-nrf52840-lock-example)
 
-zephyr_compile_options(-Werror)
-
 target_include_directories(app PRIVATE 
                            main/include 
                            ${LOCK_COMMON} 

--- a/examples/pigweed-app/nrfconnect/CMakeLists.txt
+++ b/examples/pigweed-app/nrfconnect/CMakeLists.txt
@@ -29,8 +29,6 @@ find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
 
 project(chip-nrf52840-pigweed-example)
 
-zephyr_compile_options(-Werror)
-
 target_include_directories(app PRIVATE main/include 
                            ${NRFCONNECT_COMMON}/util/include
                            ${PIGWEED_ROOT}/pw_sys_io/public

--- a/examples/shell/nrfconnect/CMakeLists.txt
+++ b/examples/shell/nrfconnect/CMakeLists.txt
@@ -26,8 +26,6 @@ find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
 
 project(chip-nrf52840-shell-example)
 
-zephyr_compile_options(-Werror)
-
 target_include_directories(app PRIVATE
                            ${CMAKE_CURRENT_SOURCE_DIR}
                            ${APP_ROOT}/shell_common/include)

